core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -351,7 +354,30 @@ public void invalidateTable(SessionContext context, TableIdentifier ident) {}
   @Override
   public Table registerTable(
       SessionContext context, TableIdentifier ident, String metadataFileLocation) {
-    throw new UnsupportedOperationException("Register table is not supported");
+    checkIdentifierIsValid(ident);
+
+    Preconditions.checkArgument(
+        metadataFileLocation != null && !metadataFileLocation.isEmpty(),
+        "Cannot register an empty metadata file location as a table");
+
+    // Throw an exception if this table already exists in the catalog.
+    if (tableExists(context, ident)) {
+      throw new AlreadyExistsException("Table already exists: %s", ident);
+    }
+
+    TableMetadata metadata = TableMetadataParser.read(io, metadataFileLocation);
+    Map<String, String> tableProperties =
+        ImmutableMap.<String, String>builder()
+            .putAll(metadata.properties())
+            .put(BaseMetastoreTableOperations.METADATA_LOCATION_PROP, metadataFileLocation)
+            .build();
+
+    return buildTable(context, ident, metadata.schema())
+        .withLocation(metadata.location())
+        .withProperties(tableProperties)
+        .withPartitionSpec(metadata.spec())
+        .withSortOrder(metadata.sortOrder())
+        .create();
```

Comment:
This discards all table data and older schemas, specs, and sort orders. I think this needs to preserve the table metadata completely.